frontend/components/NavDrawerCategoriesTree/selectors.js

```diff
@@ -21,9 +39,35 @@ export const makeGetSubcategoriesByCategoryId = () =>
     (childCategories, categoryTree, categoryId) => {
       // Check if we have to handle the root-category
       if (!categoryId && categoryTree) {
-        return categoryTree;
+        const enriched = categoryTree.map(category => {
+          const content = {content: hasCategoryContent(category)}
+          return {...category, ...content};
+        })
+        
+        return enriched;
       }
 
-      return childCategories;
+      //has category content???
+      if (childCategories) {
+        const enriched = childCategories.map(category => {
+          const content = {content: hasCategoryContent(category)}
+          return {...category, ...content};
+        });
+
+        return enriched;
+      }
     }
   );
+
+  /**
+ * Returns the selection
+ * @param {Object} state .
+ * @return {Object}
+ */
+export const getPageSwitcherSelection = (state) => {
```

Comment:
i would recommend to replace the `page-switcher` extension related selectors with

```js
/**
 * Get the state of the @shopgate-project/page-switcher extension
 * @param {Object} state Redux state
 * @returns {Function}
 */
export const getPageSwitcherState = state => state?.extensions[REDUX_NAMESPACE_SELECTION];

/**
 * Returns page switcher selection
 * @param {Object} state .
 * @return {Object}
 */
export const getPageSwitcherSelection = createSelector(
  getPageSwitcherState,
  switcherState => switcherState?.selection || {}
);
```
this gives us some caching optimization in the selectors.